tests/tests-library/es6.promise.ls

```diff
@@ -33,4 +33,32 @@ test '.race' !->
 test '.resolve' !->
   ok isFunction(core.Promise.resolve), 'Is function'
 test '.reject' !->
-  ok isFunction(core.Promise.reject), 'Is function'
\ No newline at end of file
+  ok isFunction(core.Promise.reject), 'Is function'
+test ' subclassing' !->
+  # this is ES5 syntax to create a valid ES6 subclass
+  SubPromise = (x) ->
+    self = new core.Promise(x)
+    Object.setPrototypeOf(self, SubPromise.prototype)
```

Comment:
Need add `Object.setPrototypeOf` and, for `library`, `core.Object.setPrototypeOf` existence check. IE10-